api/src/main/java/io/onetable/client/PerTableConfig.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package io.onetable.client;
+
+import java.util.List;
+
+import io.onetable.model.sync.SyncMode;
+
+public interface PerTableConfig {
+  int getTargetMetadataRetentionInHours();
+
+  String getTableBasePath();
+
+  String getTableDataPath();
+
+  String getTableName();
+
+  HudiSourceConfig getHudiSourceConfig();
+
+  List<String> getTargetTableFormats();
+
+  SyncMode getSyncMode();
+
+  String[] getNamespace();
+
+  CatalogConfig getIcebergCatalogConfig();
```

Comment:
Do we want this to just be `getCatalogConfig` and then have an Iceberg specific one? It could be possible to have a hive metastore catalog or something like that for Delta/Hudi as well

Reply:
@ashvina is planning a PerTableConfig refactor. I figured it better to leave these until then rather than change something that may need to be changed again anyway. What do you think?

Reply:
Ok I'll defer to @ashvina on this one. I am in favor of smaller change sets in general so makes sense to do that on its own when thinking about it that way.

Reply:
👍  to addressing one issue in one PR.
A good starting point would be to finalize the design of `PerTableConfig` (#297) and then proceed accordingly.

Reply:
@ashvina - I read that as agreeing to leave this as is and move on within #297 - correct?